content/en/docs/Creating derivatives/creating_bootable_images.md

```diff
@@ -38,10 +39,16 @@ FROM quay.io/luet/base:$LUET_VERSION AS luet
 FROM opensuse/leap:15.3 # or Fedora, Ubuntu
 ARG ARCH=amd64
 ENV ARCH=${ARCH}
+ENV COSIGN_EXPERIMENTAL=1 # keyless verify
+ENV COSIGN_REPOSITORY=raccos/releases-green  # repo with the signatures
 RUN zypper in -y ... # apt-get, dnf...
 
+# Here we install cosign and luet-cosign so we can verify that the images installed have been signed
+RUN luet install -y toolchain/cosign toolchain/luet-cosign
```

Comment:
Something that "puzzles" me is that in this way, when we get cosign itself and the plugin, those are not verified (via signatures, etc). So we are with the chicken and egg problem, as our first bits that we install are not verified..

Shall we maybe create a signed container where we take this tools from instead?

Reply:
how would we verify that container?

Reply:
I guess what you propose is to then we can do:

```
COPY --from=cosign /usr/bin/cosign /usr/bin/cosign
COPY --from=cosign /usr/bin/luet-cosign /usr/bin/luet-cosign
```

???

Reply:
yup, In that way if someone is _paranoic_ enough can just verify that container where it is taking the binary from

Reply:
In that case, can you just get cosign manually and verify manually beforehand?

```
# itxaka @ zeus in ~/projects/cos-toolkit-docs on git:cosign o [11:07:00] C:1
$ export COSIGN_EXPERIMENTAL=1                                                  

# itxaka @ zeus in ~/projects/cos-toolkit-docs on git:cosign o [11:10:03] 
$ export COSIGN_REPOSITORY=raccos/releases-green

# itxaka @ zeus in ~/projects/cos-toolkit-docs on git:cosign o [11:10:10] 
$ cosign verify quay.io/costoolkit/releases-green:luet-cosign-toolchain-0.0.6-1               
No TUF root installed, using embedded CA certificate.

No TUF root installed, using embedded rekor key

Verification for quay.io/costoolkit/releases-green:luet-cosign-toolchain-0.0.6-1 --
The following checks were performed on each of these signatures:
  - The cosign claims were validated
  - Existence of the claims in the transparency log was verified offline
  - Any certificates were verified against the Fulcio roots.

[{"critical":{"identity":{"docker-reference":"quay.io/costoolkit/releases-green"},"image":{"docker-manifest-digest":"sha256:9377596ed7f98c084a86a96f0536ddbfacf7a3e90567e8b577745b602d4d6d3d"},"type":"cosign container image signature"},"optional":{"Bundle":{"SignedEntryTimestamp":"MEYCIQDSN/rJG/6+EsaaYrj8wpXP5ua4ecnRQZPv1sczWOXaewIhAMpx9k7IX8N0oEZXDAg657u48JjSCHALfmRiFHHX7nbw","Payload":{"body":"eyJhcGlWZXJzaW9uIjoiMC4wLjEiLCJraW5kIjoicmVrb3JkIiwic3BlYyI6eyJkYXRhIjp7Imhhc2giOnsiYWxnb3JpdGhtIjoic2hhMjU2IiwidmFsdWUiOiI4ZGRhMmNlMzYyOGRjM2FlMmIxM2RmMDhiYjc3NTM2NjVhNzcwZDMyMGYyMjE1MzZkOTlhM2I5ODZjMWNkYzg1In19LCJzaWduYXR1cmUiOnsiY29udGVudCI6Ik1FVUNJUUMrT0Y4ZUE2NnRkRWlnRUMyM29kZ21kaTgvNisxRmdaVHE2VHk1WGs1WXVRSWdQRFhQZ2JRYUpZYlJQQUkraVZQTzVNa3QydGdGMjVXK0ozdnpUdHArdjRZPSIsImZvcm1hdCI6Ing1MDkiLCJwdWJsaWNLZXkiOnsiY29udGVudCI6IkxTMHRMUzFDUlVkSlRpQkRSVkpVU1VaSlEwRlVSUzB0TFMwdENrMUpTVVJGVkVORFFYQmxaMEYzU1VKQlowbFZRVkEwZUhaYVVXNVJSMFl6SzFaMFEySlVSbXhKVWtWTVRtMXpkME5uV1VsTGIxcEplbW93UlVGM1RYY0tTMnBGVmsxQ1RVZEJNVlZGUTJoTlRXTXliRzVqTTFKMlkyMVZkVnBIVmpKTlVrVjNSSGRaUkZaUlVVUkZkMmg2WVZka2VtUkhPWGxhVkVGbFJuY3dlUXBOVkVWNFRVUkpkMDE2VlhkTlJFSmhSbmN3ZVUxVVJYaE5SRWwzVGtSQk5VNVViR0ZOUVVGM1YxUkJWRUpuWTNGb2EycFBVRkZKUWtKblozRm9hMnBQQ2xCUlRVSkNkMDVEUVVGUlVreDBUREJhT0V0SmFGcFNXRTh2VldvNWNITkdNeXMyWWs5VFIwZ3dNakpZT1ZwMlZXNWtabFZZZDB0cmVVVkNVSEExVkdnS2FWWjFXRWRDUzNNNVRFZDZPRXhuZVhwSGNUSjVaV3g2UW1KdFMyVXpVWGx2TkVsQ2QzcERRMEZpT0hkRVoxbEVWbEl3VUVGUlNDOUNRVkZFUVdkbFFRcE5RazFIUVRGVlpFcFJVVTFOUVc5SFEwTnpSMEZSVlVaQ2QwMUVUVUYzUjBFeFZXUkZkMFZDTDNkUlEwMUJRWGRJVVZsRVZsSXdUMEpDV1VWR1J6ZEtDa2RCV2pZeVNISkRUSGRqTWt3MGQyUXpaakZCV2tObmRrMUNPRWRCTVZWa1NYZFJXVTFDWVVGR1RXcEdTRkZDUW0xcFVYQk5iRVZyTm5jeWRWTjFNVXNLUW5SUWMwMUpSMDVDWjJkeVFtZEZSa0pSWTBKQlVWTkNaMFJDSzAxSWQwZERRM05IUVZGVlJrSjZRVU5vYmtKdlpFaFNkMDlwT0haalNFcHdaRzFHTUFwYVYwNW9URmRPZG1KdVVteGlibEYwVG1wQmVscHRWVE5hVkdOMFRVUkJkMDFETUhsTmFra3pURmRLYlU1NlZYUmFhbEp0VGxkVk5FMUhVWGxQVkZVd0NreHVUakJpTTBwb1dqSlZkVm95T1haYU1uaHNXVmhDY0dONU5XcGlNakIyV1RKRmVrNXRSWGhhVkdzeVRXcFJlVmxxYkcxWk1rbDRUa1JaZGxreVJYVUtXVE5LTUUxSU9FZEJNVlZrUlZGRlFpOTNVakZOU0U5SFkxZG9NR1JJUW5wUGFUaDJXakpzTUdGSVZtbE1iVTUyWWxNNWVWbFhOV3BoUjFaNVRGaE9hQXBpYlZKcFlqTm5kbGt3T1ZSTVdGSjJZako0Y21GWVVYWk1iV1J3WkVkb01WbHBPVE5pTTBweVdtMTRkbVF6VFhaWmJsWndZa2RSZEdKWFJucGtSMVo1Q2t4WFpIbGFWMVoxVEZobk5FNXNPREpPUXpVMVdWY3hjMUZJU214YWJrMTJZVWRXYUZwSVRYWmlWMFo2WkVkV2VVMUVhMGREYVhOSFFWRlJRbWMzT0hjS1FWRkZSVXN5YURCa1NFSjZUMms0ZG1SSE9YSmFWelIxV1ZkT01HRlhPWFZqZVRWdVlWaFNiMlJYU2pGak1sWjVXVEk1ZFdSSFZuVmtRelZxWWpJd2R3cERaMWxKUzI5YVNYcHFNRVZCZDAxRVlVRkJkMXBSU1hoQlNXSnJWRWxrYmpaNmFuZHpRMmhOYW5jeVFURjNOSHB3YkVkNVpHTnRZVVo0TXpOcGFXUmFDakpDWkN0UkwxaGthbmxxV0RaSk5GWldTMU15Y2xCV1lsVkJTWGRPYkhnMmEzaDNVbFl3UjBobGIwcDZVbmd6YURseWRFcDZjR0ZwT1ROR2EzRmFUVUVLWjI5VE5YcHpkRlZwUmxsd1JXRmFLMjVrUkdWaVRVdFpSVWRhT1FvdExTMHRMVVZPUkNCRFJWSlVTVVpKUTBGVVJTMHRMUzB0Q2c9PSJ9fX19","integratedTime":1635825001,"logIndex":816654,"logID":"c0d23d6ad406973f9559f3ba2d1ca01f84147d8ffc5b8445c224f98b9591801d"}},"Subject":"https://github.com/rancher-sandbox/cOS-toolkit/.github/workflows/build-master-green-x86_64.yaml@refs/heads/master"}}]

```

Reply:
should probably add that to the docs :D

Reply:
updated!